### ISSUE TITLE
Learnable parameters are not added to the optimizer, no real training can happen!

### ISSUE THREAD

**Issue by Hippogriff** [CIntellifusion/GeometryForcing#13]
The learnable parameters of the class VGGTAlignmentLoss are never added to the optimizer. So it is unclear how the network is able to learn geometric features at all.
@CIntellifusion Please clarify where the learnable parameters of VGGTAlignmentLoss, specifically self.projectors are added to the optimizer.
**EDIT:**
The proposed algorithm has no effect on the final evaluation metrics. Simply setting the proposed algorithm to zero gives the same performance. The additional benefit comes merely by finetuning the diffusion model using diffusion loss at a very low learning rate.

**Comment by CIntellifusion** [comment]
Thanks for mention this bug! 

The projector wasn't trained in the initial commit. But the gradient can be passed back to diffusion model even the projector is not updated. Moreover, we don't need the projector during inference. So it doesn't influence the main conclusion of our paper. 

And I solved this bug by adding adding projector to the optimizer. 

Rewrite the `configure_optimizers` function in `DFotGeometryForcing` can fix this bug. 

`    def configure_optimizers(self):
        transition_params = list(self.parameters())
        print(f"[DFoTGeometryForcing] PUT Projector Into OPtimizer")
        print(f"[DFoTGeometryForcing] Number of trainable parameters: {sum(p.numel() for p in transition_params if p.requires_grad)}")
        transition_params = [p for p in transition_params if p.requires_grad]
        
        optimizer_dynamics = torch.optim.AdamW(
            transition_params,
            lr=self.cfg.lr,
            weight_decay=self.cfg.weight_decay,
            betas=self.cfg.optimizer_beta,
        )

        lr_scheduler_config = {
            "scheduler": get_scheduler(
                optimizer=optimizer_dynamics,
                **self.cfg.lr_scheduler,
            ),
            "interval": "step",
            "frequency": 1,
        }

        return {
            "optimizer": optimizer_dynamics,
            "lr_scheduler": lr_scheduler_config,
        }`

The result is slightly better than what reported in paper(FVD from 243 to 239) 

<img width="400" height="121" alt="Image" src="https://github.com/user-attachments/assets/a1b8c67c-e594-4152-828c-5522dae4e82d" />

Thanks for your detailed review on our code!

**Comment by Hippogriff** [comment]
>The projector wasn't trained in the initial commit. But the gradient can be passed back to diffusion model even the projector is not updated

Even after setting the loss to zero, I am getting the metrics that you have reported in the paper. That means the proposed loss has no effect on the training. The learning rate of 8e-6 is so small for the adapter layer to train, even if you include these layers in the optimizer. I would appreciate this ablation in the paper and also reported here.

**Comment by CIntellifusion** [comment]
Pure diffusion loss can not matching the performance of geometry forcing. We apply geometry forcing on the finetuned diffusion model, and the performance improvement is significant.  

We don't mean to train the adapter layer, we aim to train the diffusion model. 8e-6 is suitable for diffusion model finetuning. We have ablated the learning rate before from 1e-5 to 1e-6.  It is reasonable to have different learning rate for diffusion model and adapter layer, but it is not the point we want to study.

**Comment by Hippogriff** [comment]
Why do you not want to train the adapter layer? It makes no sense to propose an adaptor layer and not train it.
